client/src/App.js

```diff
@@ -35,28 +37,38 @@ library.add(
 );
 
 class App extends Component {
+  // axios configuration
+  axiosSource = axios.CancelToken.source();
+  axiosInstance = axios.create({
+    baseURL: '/api/v1/',
+    cancelToken: this.axiosSource.token
+  });
+
   state = {
     userInfo: {},
-    isLogin: true,
   };
-  // componentDidMount() {
-  //   isAuth
-  //     ? this.setState({ isLogin: true })
-  //     : this.setState({ isLogin: false });
-  // }
-  setUserInfo = user => {
-    this.setState({ userInfo: user, isLogin: true });
+
+  setUserInfo = userInfo => {
+    this.setState({ userInfo });
   };
 
   clearUserInfo = () => {
-    this.setState({ userInfo: '', isLogin: false });
+    this.setState({ userInfo: '' });
   };
 
+  componentWillUnmount() {
```

Comment:
@ashatat You might not need this, since I think App will only be unmounting when the user is leaving the page or closing the browser (which in both cases, the browser will cancel all of the page's in-flight requests).

Reply:
@tomduggan85, yeah you are right, I haven't thought about that :man_facepalming: 

@IsraaSulaiman don't forget this, please :)